src/ore/src/antichain.rs

```diff
@@ -56,8 +63,15 @@ impl<T: PartialOrder + Ord + Clone> AntichainToken<T> {
     where
         F: IntoIterator<Item = T>,
     {
-        // TODO(mjibson): Add self.antichain.less_equal(frontier) check here. Need to
-        // correctly handle the initial case where self.antichain is empty.
+        // TODO(rkhaitan): do this without the extra allocations.
+        let frontier: Vec<_> = frontier.into_iter().collect();
+        let mut new_frontier = Antichain::new();
+        new_frontier.extend(frontier.iter().cloned());
```

Comment:
i think you can extend directly from the initial iter and then use new_frontier.elements() below to only cause one new vec instead of 2.